### PR TITLE
Replace landmark correspondence classifier with text-only variant

### PR DESCRIPTION
**Pull request by efahnestock**
## Summary

Collapses the parallel `Simple*` + type-routed `landmark_correspondence_*` modules into a **single** text-only correspondence model. The old boolean/numeric/housenumber value-type routing and its 13 cross features are deleted; every tag value is encoded as text via pre-computed embeddings, and 4 cross features (text_max / text_mean / text_name / housenumber_overlap) summarize the pair.

This PR replaces the original "Add simple" commit on this branch. Class/file names drop the `Simple` prefix — historical paths (`landmark_correspondence_model.py`, `landmark_correspondence_dataset.py`, `train_landmark_correspondence.py`) are preserved with new contents.

## Changes

**Library rewrites**
- `model/landmark_correspondence_model.py` — renamed classes dropped `Simple` prefix; guards the all-padded-sample max-pool so zero-tag landmarks emit zeros (not -1e9) into the classifier.
- `data/landmark_correspondence_dataset.py` — JSONL parsing / `CorrespondencePair` preserved; type-routed encode + parse-failure-report machinery deleted. Raises `KeyError` on missing text embeddings by default (`allow_missing_text_embeddings` opts into zero fallback). `__init__` logs coverage + zero-tag pair counts. `parse_housenumber` now lives here and catches `(ValueError, TypeError, AttributeError)`. `CorrespondenceBatch.to` iterates `__dataclass_fields__`, adds `pin_memory`. `collate_correspondence` requires `text_input_dim` explicitly (bind via `functools.partial`). `include_difficulties` defaults to `("positive", "easy")`.
- `evaluation/correspondence_matching.py` — deleted the type-routed path and `build_correspondence_similarity_matrix`. `precompute_raw_cost_data` is the single inference entry point; `similarity_from_raw_data` consumes its artifact. Added a GPU-vectorized 4-feature cross-feature path (replaces the per-pair Python loop) and a small `compute_pairs_cost_matrix` helper for interactive tools.

**Scripts**
- `train_landmark_correspondence.py` — `best_auc` initializes to `-inf`; `compute_metrics` only swallows `ValueError` when a single label class is present; warns if no best checkpoint was ever saved.
- `export_correspondence_similarity.py` — single path. Always saves the raw artifact; `--from_raw` loads an existing one; `--compute_similarity` additionally writes a similarity matrix.
- `panorama_viewer.py`, `evaluate_correspondence_model.py`, `inspect_correspondence_model.py` (marimo) — ported to the new forward signature and `collate_correspondence` + `functools.partial` pattern.
- `precompute_value_embeddings.py` — drops the `--all_value_types` flag; every tag value keyed by `TAG_KEY_TO_IDX` is embedded.

**Tests**
- `model/landmark_correspondence_model_test.py` — rewritten: encoder shape, all-padded zero output, mask invariance, classifier shape.
- `data/landmark_correspondence_dataset_test.py` — rewritten: `parse_housenumber` (incl. `"512D"`, `None`, `int`), strict-vs-opt-in missing embeddings, 4-feature cross layout, `_pad_side`, mixed empty/real collation, dataset difficulty filtering, `CorrespondenceBatch.to`.
- `evaluation/correspondence_matching_test.py` — new. `precompute_raw_cost_data` shape/index contract against a hand-built fake VigorDataset; `compute_pairs_cost_matrix` shape/range; GPU vs. Python-loop cross-feature parity.

## Review findings addressed

From the four review agents that commented on the original PR (code-reviewer, silent-failure-hunter, type-design-analyzer, pr-test-analyzer):

- Crash on `--simple` without `--save_raw` (collapsed to one path)
- Silent zero-fallback on missing text embeddings (raises by default + coverage log)
- `CorrespondenceBatch.to` using `__dict__` (now `__dataclass_fields__`); added `pin_memory`
- `collate_correspondence` hardcoded `text_dim = 768` (now a required arg)
- `include_difficulties` default divergence
- `parse_housenumber` too-narrow exception handling
- All-padded-sample max-pool -1e9 leak
- Python-loop cross features (vectorized on GPU)
- Dead imports, mid-file imports, duplicated constants
- `best_auc = 0.0` never-saved edge case
- `compute_metrics` swallowing all `ValueError`s
- Missing tests (three new test files)

## Test plan

- [x] `bazel build //experimental/overhead_matching/swag/...` passes
- [x] `bazel test //experimental/overhead_matching/swag/...` — all 31 tests pass, including the three new/rewritten correspondence test files
- [ ] Train `train_landmark_correspondence` for a few epochs on a small dataset; confirm loss decreases, `best_model.pt` is written, `include_difficulties` filter applies
- [ ] Run `export_correspondence_similarity` on one city; confirm raw artifact saved. Re-run with `--from_raw --compute_similarity` and verify the similarity matrix is identical to a single-shot run.
- [ ] Intentionally pass a partial text embeddings pickle; confirm `KeyError` at dataset construction with coverage summary.

### CODE REVIEW

File: experimental/overhead_matching/swag/scripts/panorama_viewer.py [hunk 1]
```diff
@@ -2985,10 +2985,11 @@ def get_correspondence_details(index, sat_idx):
                         'osm_landmarks': []})
 
     try:
-        cost_matrix = compute_cost_matrix(
+        cost_matrix = compute_pairs_cost_matrix(
             pano_tags_list, osm_tags_list, CORRESPONDENCE_MODEL,
             CORRESPONDENCE_TEXT_EMBEDDINGS, CORRESPONDENCE_TEXT_INPUT_DIM,
             CORRESPONDENCE_DEVICE,
+            allow_missing_text_embeddings=True,
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/scripts/panorama_viewer.py]: set this to false

File: experimental/overhead_matching/swag/evaluation/correspondence_matching.py [hunk 2]
```diff
@@ -587,16 +104,15 @@ def _aggregate_score(
         weights = [1.0] * len(match_probs)
     if aggregation == AggregationMode.SUM:
         return sum(p * w for p, w in zip(match_probs, weights))
-    elif aggregation == AggregationMode.MAX:
+    if aggregation == AggregationMode.MAX:
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/evaluation/correspondence_matching.py]: why swap here?

File: experimental/overhead_matching/swag/evaluation/correspondence_matching.py [hunk 3]
```diff
@@ -690,37 +211,319 @@ def match_and_aggregate(
     )
 
 
+# ---------------------------------------------------------------------------
+# Encoder / cross-feature helpers used by the precompute path
+# ---------------------------------------------------------------------------
+
+def _pad_encoded(encoded_list: list[dict], text_input_dim: int) -> dict:
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/evaluation/correspondence_matching.py]: redefining it from elsewhere?

File: experimental/overhead_matching/swag/evaluation/correspondence_matching.py [hunk 4]
```diff
@@ -690,37 +211,319 @@ def match_and_aggregate(
...
+            logits = model.classify_from_reprs(
+                pano_exp[s:e].to(device),
+                osm_exp[s:e].to(device),
+                cross_tensor[s:e].to(device),
+            ).squeeze(-1)
+            chunks.append(torch.sigmoid(logits).cpu().numpy())
+    return np.concatenate(chunks).reshape(n_pano, n_osm)
+
+
+def _compute_cross_features_for_pairs(
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/evaluation/correspondence_matching.py]: move this to the test then
